contracts/test/issue.js

```diff
@@ -0,0 +1,284 @@
+                blockHeight = await ethers.provider.getBlockNumber()
+
+                filterMsg = await getFilterMsgByFilter(
+                    {
+
+                        "fromBlock.pending": {
+                            'fromBlock': 'pending'
+                        },
+                        "fromBlock.blockHeight+1000": {
+                            'fromBlock': BigNumber.from(blockHeight).add(1000).toHexString().replace('0x0', '0x')
+                        },
+
+                        "toBlock.earliest": {
+                            "toBlock": "earliest"
+                        },
+                    }, 3)
+            });
+
+
+            describe("fromBlock", function () {
+
+
+                it.skip("pending,should return error msg", async () => {
+                    //invalid from and to block combination: from > to
+                    expect(filterMsg["fromBlock.pending"].error).to.be.not.equal(undefined)
+
+                })
+
+
+                it.skip("blockNumber(blockHeight+1000),should return 0 log", async () => {
+
+                    expect(filterMsg["fromBlock.blockHeight+1000"].logs.length).to.be.equal(0)
+                })
+
+            })
+
+            describe('toBlock', function () {
+
+                it.skip("earliest,should return error msg", async () => {
+                    //invalid from and to block combination: from > to
+                    expect(filterMsg["toBlock.earliest"].error).to.be.not.equal(undefined)
+                })
+
+            });
+
+
+        });
+
+        describe('filter topic', function () {
+
+            let contractAddress;
+
+            let topic0 = "0x0000000000000000000000000000000000000000000000000000000000000001";
+            let topic1 = "0x0000000000000000000000000000000000000000000000000000000000000002";
+            let topic2 = "0x0000000000000000000000000000000000000000000000000000000000000003";
+            let topic3 = "0x0000000000000000000000000000000000000000000000000000000000000004";
+            let filterMsgMap;
+            let logContract;
+            let blockHeight;
+
+            before(async function () {
+
+                blockHeight = await ethers.provider.getBlockNumber()
+                //deploy contract
+                let logContractInfo = await ethers.getContractFactory("logContract");
+                logContract = await logContractInfo.deploy()
+                await logContract.deployed()
+                contractAddress = logContract.address
+                let topicsMap = {
+
+                    "topic.[[A, B],[A, B]].yes": {
+                        "topics": [[topic3, topic0], [null, null, topic2]]
+                    },
+                    "topic.[[A, B],[A, B]].no": {
+                        "topics": [[topic0, topic2, topic3], [null, topic2], [topic1]]
+                    },
+                }
+
+                filterMsgMap = await getTopicFilter(topicsMap, logContract, 10)
```

Comment:
This test case contains multiple transactions, do you think it should be used for mainnet regression testing?
If not, maybe you can add conditional code like this:
```javascript
const { isGwMainnetV1 } = require('../utils/network');

describe('...', () => {
  if (isGwMainnetV1()) {
    return;
  }

  ...
});
```

Reply:
With conditional code  added, the main net will not run ;
Add skip type , skip can be removed after the bug is fixed

Reply:
> - https://github.com/nervosnetwork/godwoken-web3/issues/451

Should we add this issue to the test case as a reminder?